### PR TITLE
feat(log): add global setting for log level

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: [longhorn/longhorn#5888](https://github.com/longhorn/longhorn/issues/5888)

add global setting for log level used in longhorn manager, default is warning

### CODE REVIEW

**Comment by ChanYiLin** [comment]
Screenshot from 2023-05-18 17-03-13

update log level by setting, there is no info msg after update log level to warning

File: types/setting.go [hunk 1]
```diff
@@ -1112,6 +1115,16 @@ var (
 		ReadOnly:    false,
 		Default:     "5",
 	}
+
+	SettingDefinitionLogLevel = SettingDefinition{
+		DisplayName: "Log Level",
+		Description: "The log level 0 (Panic) ~ 6 (Trace) used in longhorn manager. By default 3 (Warning).",
```
Comment (shuo-wu) [inline types/setting.go:1121]: Should the default level be `warning` rather than `info`?

**Review by shuo-wu — COMMENTED** [review]
How do longhorn manager pods load this setting during startup?

File: types/setting.go (hunk 1 above)
Comment (innobead) [reply types/setting.go:1121]: `info` is lower than `warning`, so `warning` will have less logs than info level.

Comment (innobead) [inline types/setting.go:1121]: Let's use level constant instead of number. You can use `ParseLevel` to achieve that.

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. 

But, please use log string intead of log number to specify log level.

**Comment by innobead** [comment]
> How do longhorn manager pods load this setting during startup?

@ChanYiLin I think we can do something during the start?

File: types/setting.go (hunk 1 above)
Comment (shuo-wu) [reply types/setting.go:1121]: `info` logs contain some important info like object state change. Suppressing it by default would make the support bundle debugging more difficult.

Comment (innobead) [reply types/setting.go:1121]: @shuo-wu do you mean it will be more tedious? however, right now it's `debug` level which is even lower than `info`.

Comment (ChanYiLin) [reply types/setting.go:1121]: I agree with @shuo-wu 's comment
Most of the people use default setting
So if they generate support bundle when encountering some issue, there will be no info message in the log and make us more difficult to debug

**Comment by ChanYiLin** [comment]
> > How do longhorn manager pods load this setting during startup?
> 
> @ChanYiLin I think we can do something during the start?

Longhorn-manager setup the setting at the very beginning
https://github.com/longhorn/longhorn-manager/blob/master/app/daemon.go#LL173C1-L173C27

It generates the setting CR and trigger the setting controller
https://github.com/longhorn/longhorn-manager/blob/master/datastore/longhorn.go#L74

There might have some debug level message before controller starts, but there is a PR to disable debug args
https://github.com/longhorn/longhorn/pull/5858

Plus we are using Info as default, so I think we don't need to do anything here

File: types/setting.go (hunk 1 above)
Comment (ChanYiLin) [reply types/setting.go]: Fixed, thanks

Comment (ChanYiLin) [reply types/setting.go]: Change to default info level

Comment (innobead) [reply types/setting.go]: I thought it was info. Yes we should use info.

File: types/setting.go [hunk 2]
```diff
@@ -1112,6 +1115,16 @@ var (
...
 	}
+
+	SettingDefinitionLogLevel = SettingDefinition{
+		DisplayName: "Log Level",
+		Description: "The log level Panic, Fatal, Error, Warn, Info, Debug, Trace used in longhorn manager. By default Info.",
+		Category:    SettingCategoryGeneral,
+		Type:        SettingTypeString,
+		Required:    true,
+		ReadOnly:    false,
+		Default:     "Info",
```
Comment (innobead) [inline types/setting.go:1126]: This is case insensitive.

**Review by innobead — APPROVED** [review]
LGTM. Remember to update longhorn/longhorn manifests.

File: types/setting.go (hunk 2 above)
Comment (ChanYiLin) [reply types/setting.go:1126]: Yes, ParseLevel will lower the string

**Pull request merged**